### PR TITLE
Pick pods for preemption based on StartTime of pods when priorities a…

### LINKED ISSUE #72039: Scheduler preemption: take pod's StartTime into account when select potentialVictims
**What would you like to be added**:
When selecting victims on node in `selectVictimsOnNode`(https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/core/generic_scheduler.go#L982), take pod's `StartTime` into account. When two pod's Priority is equal, preempt pod whose `StartTime` is later first. 

**Why is this needed**:
 If a pod's `StartTime` has been running for a longer time, it might finish earlier, or it might have done more things. It might make more sense to preempt them later, e.g. a user submitted a job A at time 9:00 am,  then he submitted a job B at time 10:00 am, he might expect that scheduler preempts pods for job B first.

/kind feature

### PR DESCRIPTION
**Pull request by hex108**
…re equal

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Pick pods for preemption based on StartTime of pods when their priorities are equal.

**Which issue(s) this PR fixes**:

Fixes #72039

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig scheduling

### CODE REVIEW

**Review by wgliang — APPROVED** [review]
/lgtm

File: pkg/scheduler/util/utils.go [hunk 1]
```diff
@@ -135,8 +135,14 @@ func (l *SortableList) Sort() {
 }
 
 // HigherPriorityPod return true when priority of the first pod is higher than
-// the second one. It takes arguments of the type "interface{}" to be used with
-// SortableList, but expects those arguments to be *v1.Pod.
+// the second one. If two pods' priority are equal, compare their StartTime.
```
Comment (denkensk) [inline pkg/scheduler/util/utils.go]: pods' priority --> pods's priorities

File: pkg/scheduler/util/utils.go [hunk 2]
```diff
@@ -135,8 +135,14 @@ func (l *SortableList) Sort() {
...
+// It takes arguments of the type "interface{}" to be used with SortableList,
+// but expects those arguments to be *v1.Pod.
 func HigherPriorityPod(pod1, pod2 interface{}) bool {
-	return GetPodPriority(pod1.(*v1.Pod)) > GetPodPriority(pod2.(*v1.Pod))
+	p1 := GetPodPriority(pod1.(*v1.Pod))
+	p2 := GetPodPriority(pod2.(*v1.Pod))
+	if p1 != p2 {
+		return p1 > p2
+	}
+	return GetPodStartTime(pod1.(*v1.Pod)).Before(GetPodStartTime(pod2.(*v1.Pod)))
```
[the remark this answers is not in the document]
Comment (hex108) [reply pkg/scheduler/util/utils.go]: Thanks. Is there any difference except code style? The former might be more clear. Open to other options. :)

File: pkg/scheduler/util/utils.go (hunk 1 above)
Comment (hex108) [reply pkg/scheduler/util/utils.go]: In `pods'`, there has already a 's' in the last letter, so the last `s` of `pods's` could be omitted?  `priorities` addressed.

File: pkg/scheduler/core/generic_scheduler_test.go [hunk 3]
```diff
@@ -946,6 +946,19 @@ func TestSelectNodesForPreemption(t *testing.T) {
 				{ObjectMeta: metav1.ObjectMeta{Name: "e", UID: types.UID("e")}, Spec: v1.PodSpec{Containers: largeContainers, Priority: &highPriority, NodeName: "machine2"}}},
 			expected: map[string]map[string]bool{"machine1": {"b": true, "c": true}},
 		},
+		{
+			name:       "mixed priority pods are preempted, pick latter StartTime one when priorities are equal",
```
Comment (bsalamat) [inline pkg/scheduler/core/generic_scheduler_test.go]: s/latter/later/

File: pkg/scheduler/util/utils.go [hunk 4]
```diff
@@ -135,8 +135,14 @@ func (l *SortableList) Sort() {
 }
 
 // HigherPriorityPod return true when priority of the first pod is higher than
-// the second one. It takes arguments of the type "interface{}" to be used with
-// SortableList, but expects those arguments to be *v1.Pod.
+// the second one. If two pods' priorities are equal, compare their StartTime.
+// It takes arguments of the type "interface{}" to be used with SortableList,
+// but expects those arguments to be *v1.Pod.
```
Comment (bsalamat) [inline pkg/scheduler/util/utils.go]: This change is fine, but we should rename this function to something like `MoreImportantPod` as it checks more than just priority.

File: pkg/scheduler/core/generic_scheduler_test.go (hunk 3 above)
Comment (hex108) [reply pkg/scheduler/core/generic_scheduler_test.go]: Addressed.

File: pkg/scheduler/util/utils.go (hunk 4 above)
Comment (hex108) [reply pkg/scheduler/util/utils.go]: Addressed.

**Comment by hex108** [comment]
Updated. PTAL @bsalamat :)

**Review by bsalamat — COMMENTED** [review]
/lgtm
/approve

Thanks, @hex108!

**Comment by tongchao199** [comment]
👍

**Pull request merged**